src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -599,6 +599,15 @@ bool tryCalculateRequiredMembers(out ImmutableSegmentedDictionary<string, Symbol
                 var baseAllRequiredMembers = BaseTypeNoUseSiteDiagnostics?.AllRequiredMembers ?? ImmutableSegmentedDictionary<string, Symbol>.Empty;
                 var hasDeclaredRequiredMembers = HasDeclaredRequiredMembers;
 
+                // Fast check: avoid realizing GetMembersUnordered() for substituted symbols when the original
+                // definition doesn't contain any required members or any members with the same name as a required
+                // member from the base type.
+                if (!ReferenceEquals(OriginalDefinition, this)
+                    && !OriginalDefinition.GetMembersUnordered().Any(static (member, baseAllRequiredMembers) => member.IsRequired() || baseAllRequiredMembers.ContainsKey(member.Name), baseAllRequiredMembers))
```

Comment:
>&& !OriginalDefinition.GetMembersUnordered().Any(static (member, baseAllRequiredMembers) => member.IsRequired() || baseAllRequiredMembers.ContainsKey(member.Name), baseAllRequiredMembers)) [](http://example.com/codeflow?start=20&length=188)

Rather than trying to maintain matching logic between this check and the loop below, consider aways iterating over ```OriginalDefinition.GetMembersUnordered``` in the loop and using ```member.SymbolAsMember(this)``` helper to "project" the members to the current type when it is really necessary (the override check, add to ```requiredMembersBuilder```, etc.) We should be able to enforce that the "projection" is performed just once.

Reply:
➡️ This simplification has now been applied.